src/handlers/coins.ts

```diff
@@ -0,0 +1,57 @@
+import { FastifyRequest } from "fastify";
+
+import { Logger } from "../logger";
+import { AuthenticateRequestParameters, AuthenticationService } from "../services/authentication";
+import { UserService } from "../services/user";
+
+const logger: Logger = new Logger({
+    name: "OsuCoinsHandler",
+});
+
+interface UserCoinsParameters extends AuthenticateRequestParameters {
+    action: "check" | "use" | "recharge" | "earn";
+    // count
+    c: number;
+    // checksum (username + count + osuycoins)
+    cs: string;
+}
+
+export const getUserCoins = async (
+    request: FastifyRequest<{ Querystring: UserCoinsParameters }>,
+): Promise<number> => {
+    // i need me them typehints or i kms
```

Comment:
they're implied anyways what are you doing

Reply:
they were typed as any for me

Reply:
now it fixed itself nice great